Spanner/src/RequestHeaderTrait.php

```diff
@@ -0,0 +1,80 @@
+<?php
+/**
+ * Copyright 2024 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Spanner;
+
+use Google\Cloud\Spanner\Session\SessionPoolInterface;
+
+/**
+ * Shared methods for request headers.
+ */
+trait RequestHeaderTrait
+{
+    /**
+     * Add the `x-goog-spanner-route-to-leader` header value to the request.
+     *
+     * @param array $args
+     * @param bool $value
+     * @param string $context
+     * @return array
+     */
+    private function addLarHeader(
+        array $args,
+        bool $value = true,
+        string $context = SessionPoolInterface::CONTEXT_READWRITE
+    ) {
+        // If value is false, unset LAR header.
+        if (!$value) {
+            return $this->conditionallyUnsetLarHeader($args, $value);
+        }
+        // If value is true and context is READWRITE, set LAR header.
+        if ($context === SessionPoolInterface::CONTEXT_READWRITE) {
+            $args['headers']['x-goog-spanner-route-to-leader'] = ['true'];
```

Comment:
Would like to see this header string as a const/static value.

Why is `true` passed as a string?

Reply:
Sending the value as a `bool` throws error from the backend, Also, Spanner team confirmed it needs to be send as `string`. Added a new `const` header in this file. thanks!

Reply:
Making it as `const` in trait throws the following error in our tests, let me try with `static`:
`PHP Fatal error:  Traits cannot have constants`